examples/story-book/story-book.gpt

```diff
@@ -0,0 +1,47 @@
+tools: story-writer, story-illustrator, sys.write, sys.read, sys.download, open
+description: Writes a children's book and generates illustrations for it.
+args: story: The story to write and illustrate. Can be a prompt or a complete story.
+args: pages: The number of pages to generate
+
+Do the following steps sequentially:
+
+1. If ${story} is a prompt and not a complete children's story, call story-writer
+   to write a story based on the prompt.
+3. Take ${story} and break it up into ${pages} logical "pages" of text.
+3. For each page:
+   - Call story-illustrator to illustrate it.
+   - Download the illustration to a file at examples/story-book/pages/<page_number>.png.
+4. For each page and its illustration write an html file that:
+   - Use examples/story-book/pages/template.html as an exact guide
+     - Make sure the format and tags are the same
+     - Include the exact same <style> tag
+     - All text content is written in a <p> tag and the image is in an <img> tag.
+   - Is created at examples/story-book/pages/page<page number>.html
+5. Edit the "pages" array in examples/story-book/index.html to serve the pages you created.
+6. Open examples/story-book/index.html
+
+---
+name: story-writer
+description: Writes a story for children
+args: prompt: The prompt to use for the story
+
+Write a story with a tone for children based on ${prompt}.
+
+---
+name: story-illustrator
+tools: ../../../image-generation/tool.gpt
+description: Generates a illustration for a children's story
+args: text: The text of the page to illustrate
+
+Think of a good prompt to generate an image to represent $text. Make sure to
+include the name of any relevant characters in your prompt. Then use that prompt to 
+generate an illustration. Append any prompt that you have with "in a children's book
+cartoon style". Only return the URL of the illustration.
+
+---
+name: open
+description: Opens a given file
+args: file-path: Path to the file to open
+
+#!/bin/bash
+open examples/story-book/index.html
```

Comment:
Does the `open` command exist in Linux? I think this might be a macOS-only thing.

Reply:
Looks like for Linux it needs to be xdg-open